### ISSUE TITLE
Error if empty useragent

### ISSUE THREAD

**Issue by jordipolo87** [matomo-org/device-detector#6697]
Hello guys,

Thank you for this nice library and all your work. :)

A few days ago we have updated the library to the version 4.2.1 and we are seeing this error when the useragent is empty:
_Return value of DeviceDetector\DeviceDetector::getUserAgent() must be of the type string, null returned_

Maybe this method should allow a null return type if the useragent is not set or is empty?

Best regards,
Jordi

**Comment by sanchezzzhak** [comment]
Hi. 4.2.1 contains a  micro-bug
I recommend updating to 4.2.2

as a workaround, you can use the following code
```php
$useragent = $_SERVER['HTTP_USER_AGENT'] ?? '';
 $dd = new DeviceDetector;
$dd->setUserAgent($useragent);
$dd->parse();
```

**Comment by sanchezzzhak** [comment]
The only thing that can come to mind is that there is some DI container that does not pass parameters to the class DeviceDetector.
```
    public function testIsUserAgentEmptyString()
    {
        $dd = new DeviceDetector;
        $dd->parse();
        $this->assertEquals('', $dd->getUserAgent());
    }
```
and then we call the getUserAgent() method and get an error

 expected error: `TypeError : Return value of DeviceDetector\DeviceDetector::getUserAgent() must be of the type string, null returned`

**Comment by sgiehl** [comment]
@sanchezzzhak haven't looked in detail yet, but guess if useragent needs to be a string it could cause problems when `setUserAgent` is called with any other variable type, or it isn't called at all maybe...

**Comment by sgiehl** [comment]
fixed with 3fcceafc601444412ce65297ba4ce878ee77e450
